woocommerce/woocommerce
# Skip tax calculation for shipping when its status is set as none

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Order totals calculation didn't respect a shipping method tax status, this PR enables it, as well as adding in all the places it needs to be.

It seems this PR changed also changed the output of `wc/v3/orders` `shipping_lines` tax status, I'm not sure if that should be shown there or not, or if this PR needs extra measures to account for that. I can look into removing it if needed.

Closes https://github.com/woocommerce/woocommerce/issues/44356



### How to test the changes in this Pull Request:

1. Enable taxes, make sure they account for shipping, set them to a predictable amount (20% for example).
2. Add 2 flat rates, both having the same price, one of them should have a tax status of none, the other should be taxable.
3. Add an item to cart and go to Checkout, select the taxable shipping rate, ensure the totals are correct `(item + shipping) * 120%`.
4. Place the order, ensure the total value is still the same in the order confirmation screen.
5. Add the same item back to cart, go to Checkout, select the none-taxable item, make sure the totals are correct `(item * 120%) + shipping`.
6. Place the order,  ensure the total value is still the same in the order confirmation screen.
7. Go to the admin order screen, and go to the first order, make sure the "Recalculate" button is visible, if it's not, change the order status to `on hold` and save.
8. Take note of the totals, hit "Recalculate", the totals should stay the same.
9. Go to the newer order, do the same, ensure the values are stable.



### Changelog entry




<details>

#### Significance




#### Type




#### Message
Skip capturing shipping taxes if the shipping method tax status is set to none.

#### Comment 

</details>
